app/javascript/components/UI/Menu.js

```diff
@@ -1,26 +1,231 @@
+            percent: 5,
+          };
+        }
+      case 'meme-selection':
+        if (activeImg === true) {
+          return {
+            src: complete5_10_act,
+            percent: 10,
+          };
+        } else {
+          return {
+            src: complete5_10,
+            percent: 5,
+          };
+        }
+      case 'selected-giphy-follow':
+        if (activeImg === true) {
+          return {
+            src: complete15_act,
+            percent: 15,
+          };
+        } else {
+          return {
+            src: complete15,
+            percent: 15,
+          };
+        }
+      case 'emotion-intensity':
+        if (activeImg === true) {
+          return {
+            src: complete20_act,
+            percent: 20,
+          };
+        } else {
+          return {
+            src: complete20,
+            percent: 20,
+          };
+        }
+      case 'productivity-check':
+        if (activeImg === true) {
+          return {
+            src: complete25_act,
+            percent: 25,
+          };
+        } else {
+          return {
+            src: complete25,
+            percent: 25,
+          };
+        }
+      case 'productivity-bad-follow-up':
+        if (activeImg === true) {
+          return {
+            src: complete35_act,
+            percent: 35,
+          };
+        } else {
+          return {
+            src: complete35,
+            percent: 35,
+          };
+        }
+      case 'causes-to-celebrate':
+        if (activeImg === true) {
+          return {
+            src: complete45_act,
+            percent: 45,
+          };
+        } else {
+          return {
+            src: complete45,
+            percent: 45,
+          };
+        }
+      case 'recognition':
+        if (activeImg === true) {
+          return {
+            src: complete65_act,
+            percent: 65,
+          };
+        } else {
+          return {
+            src: complete65,
+            percent: 65,
+          };
+        }
+      case 'icebreaker-answer':
+        if (activeImg === true) {
+          return {
+            src: complete85_act,
+            percent: 85,
+          };
+        } else {
+          return {
+            src: complete85,
+            percent: 85,
+          };
+        }
+      case 'icebreaker-question':
+        if (activeImg === true) {
+          return {
+            src: complete90_act,
+            percent: 90,
+          };
+        } else {
+          return {
+            src: complete90,
+            percent: 90,
+          };
+        }
+        //TODO check after the merge to master branch
+      case 'results':
+        if (activeImg === true) {
+          return {
+            src: complete100_act,
+            percent: 100,
+          };
+        } else {
+          return {
+            src: complete100,
+            percent: 100,
+          };
+        }
+        default:
+          return "";
+        }
```

Comment:
Using a lookup object is preferable to a long conditional because it's easier to read and change. I ran this block through ChatGPT4 and asked about it, and it suggested something like this:
```javascript
const segmentsMap = {
  'emotion-selection-web': { src: 'complete0', percent: 0 },
  'emotion-entry': { src: 'complete5_10', percent: 5 },
  'meme-selection': { src: 'complete5_10', percent: 10 },
  'selected-giphy-follow': { src: 'complete15', percent: 15 },
  'emotion-intensity': { src: 'complete20', percent: 20 },
  'productivity-check': { src: 'complete25', percent: 25 },
  'productivity-bad-follow-up': { src: 'complete35', percent: 35 },
  'causes-to-celebrate': { src: 'complete45', percent: 45 },
  'recognition': { src: 'complete65', percent: 65 },
  'icebreaker-answer': { src: 'complete85', percent: 85 },
  'icebreaker-question': { src: 'complete90', percent: 90 },
  'results': { src: 'complete100', percent: 100 },
};

if (segmentsMap[lastSegment]) {
  const { src, percent } = segmentsMap[lastSegment];
  return {
    src: activeImg ? `${src}_act` : src,
    percent: percent,
  };
} else {
  return "";
}

```